Weather/Modules/CityDetail/Interface/CityDetailInterfaces.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+import UIKit
+
+
+/// The idea behind keeping all protocol declarations for all VIPER module building blocks
+/// (Router, Builder, Presenter, Interactor, ViewModel and View) in one file
+/// is to make it more explicit that they work together to construct a VIPER module.
+
+
+// ------------------------------- //
+// -- Builder:
+// ------------------------------- //
+
+protocol WeatherDetailBuilder {
+    
+    func buildWeatherDetailModule(city: String) -> UIViewController?
+}
+
+
+// ------------------------------- //
+// -- Interactor:
+// ------------------------------- //
+
+enum FetchCityWeatherResult {
+    
+    case Success(weather: WeatherData)
+    case Failure(reason: NSError)
+}
+
+protocol WeatherDetailInteractor {
+    
+    func fetchCityWeather(city: String, completion: (FetchCityWeatherResult) -> ())
+}
+
+
+// ------------------------------- //
+// -- View Models:
+// ------------------------------- //
+
+struct WeatherDetailViewModel {
+    
+    let cityName    : String
+    let temperature : String
+    let forecasts   : [WeatherDetailForecastViewModel]
+}
+
+struct WeatherDetailForecastViewModel {
+    let day : String
+    let temp: String
+}
+
+
+// ------------------------------- //
+// -- Presenter:
+// ------------------------------- //
+
+protocol WeatherDetailPresenter: class {
+    
+    var         city        : String                    {get}
+    
+    /// Required dependencies for that presenter in VIPER architecture.
+    /// The idea is to make it explicit in protocol what other components
+    /// are needed for this module presenter to function properly.
+    // ???: Is that a good idea to declare those dependencies in protocol,
+    // ???: or should they be a part of contrete implementation instead?
+    
+    var         interactor  : WeatherDetailInteractor   {get set}
+    unowned var view        : WeatherDetailView         {get}
```

Comment:
The unowned here is interesting. It was suggested to me during the iOSCon presentation and I forgot :)
Thanks for raising again!

There are a couple of considerations though:
1. Unowned vs weak has the advantage of not being optional, and on the converse side, it won't nil out the pointer in case the unowned reference actually disappears, so dangerous. Unless you are sure it won't disappear!
2. I believe it could be made "let" as you anyway have it in your initializer and we don't expect it to change.

So probably the best solution is:

unowned let view

which requires it mandatory in the initializer.

Will spike this but I currently don't see issue with that, but simplification on the view usage itself.